src/utils.ts

```diff
@@ -83,9 +95,11 @@ export const toMessage = (message: PubSubRPCMessage): Message => {
     type: 'signed',
     from: peerIdFromBytes(message.from),
     topic: message.topic ?? '',
-    sequenceNumber: bigIntFromBytes(message.sequenceNumber),
+    sequenceNumber: bigIntFromBytes(message.sequenceNumber ?? new Uint8Array(0)),
     data: message.data ?? new Uint8Array(0),
-    signature: message.signature,
+    // @ts-expect-error this is checked in `isSigned`
+    signature: message.signature ?? new Uint8Array(0),
+    // @ts-expect-error key need not be defined
     key: message.key
```

Comment:
Instead of `ts-expect-error`, please verify that the key is present and correct - e.g. it should come from the message or the peer id, and if it's present in both the message and the peer id, they should be the same.

Reply:
Fixed.

As far as secp256k1 or ed25519 keys are concerned, consider the current behaviour [here](https://github.com/libp2p/go-libp2p-pubsub/blob/973fef56e1e6fd6afc568ea3c1c2f654f7754d49/sign.go#L109). Since the public key is extractable, the `msg.Key` field is left unset, and the current js implementation considers this as an unsigned message.